validate address panic

Calling validate address using a 1 of 2 multisig address on testnet.

```
2020-03-14 01:16:06.893 [INF] RPCS: RPC method walletinfo invoked by 127.0.0.1:36882
2020-03-14 01:16:06.973 [INF] RPCS: RPC method accountsyncaddressindex invoked by 127.0.0.1:36882
2020-03-14 01:16:08.345 [INF] RPCS: RPC method validateaddress invoked by 127.0.0.1:36882
panic: interface conversion: *udb.scriptAddress is not udb.ManagedPubKeyAddress: missing method Index

goroutine 898 [running]:
decred.org/dcrwallet/wallet.(*managedP2PKHAddress).PubKey(0xc00007bac0, 0xfd6f40, 0xc00007bac0, 0x7f6050128468)
        /home/joe/git/dcrwallet/wallet/addresses.go:156 +0x44
decred.org/dcrwallet/internal/rpc/jsonrpc.(*Server).validateAddress(0xc0000b2b40, 0xfd4d40, 0xc000aa4150, 0xbaf540, 0xc000506570, 0xbaf540, 0xc000506570, 0x0, 0x0)
        /home/joe/git/dcrwallet/internal/rpc/jsonrpc/methods.go:3569 +0x3a1
decred.org/dcrwallet/internal/rpc/jsonrpc.lazyApplyHandler.func2(0x0, 0x0, 0x0)
        /home/joe/git/dcrwallet/internal/rpc/jsonrpc/methods.go:238 +0x193
decred.org/dcrwallet/internal/rpc/jsonrpc.(*Server).websocketClientRespond.func1(0xc0005020c8, 0xc00056b9c0, 0xc00007b740, 0xc0000ac320, 0xfd4d40, 0xc000aa4150)
        /home/joe/git/dcrwallet/internal/rpc/jsonrpc/server.go:436 +0xc2
created by decred.org/dcrwallet/internal/rpc/jsonrpc.(*Server).websocketClientRespond
        /home/joe/git/dcrwallet/internal/rpc/jsonrpc/server.go:433 +0x34e
```

--- a/internal/rpc/jsonrpc/methods.go
+++ b/internal/rpc/jsonrpc/methods.go
@@ -3575,7 +3575,7 @@ func (s *Server) validateAddress(ctx context.Context, icmd interface{}) (interfa
 		result.PubKeyAddr = pubKeyAddr.String()
 	case wallet.P2SHAddress:
 		result.IsScript = true
-		script, _ := ka.RedeemScript()
+		_, script := ka.RedeemScript()
 		result.Hex = hex.EncodeToString(script)
 
 		// This typically shouldn't fail unless an invalid script was

--- a/wallet/addresses.go
+++ b/wallet/addresses.go
@@ -205,7 +205,7 @@ func (w *Wallet) KnownAddress(ctx context.Context, a dcrutil.Address) (KnownAddr
 		addr: ma,
 	}
 	var p2pkhKnownAddr managedP2PKHAddress
-	var p2shKnownAddr managedP2PKHAddress
+	var p2shKnownAddr managedP2SHAddress
 	var child uint32
 	switch ma := ma.(type) {
 	case udb.ManagedPubKeyAddress:

--- a/wallet/addresses.go
+++ b/wallet/addresses.go
@@ -105,7 +105,7 @@ type P2SHAddress interface {
 	// version is the script version of the address, or the script version
 	// of the redeemed previous output, and must be used for any operations
 	// involving the script.
-	RedeemScript() (script []byte, version uint16)
+	RedeemScript() (version uint16, script []byte)
 }
 
 // managedAddress implements KnownAddress for a wrapped udb.ManagedAddress.
@@ -172,7 +172,10 @@ func (m *managedBIP0044Address) Path() (account, branch, child uint32) {
 // managedP2SHAddress implements P2SHAddress for a wrapped udb.ManagedAddress.
 type managedP2SHAddress struct {
 	managedAddress
-	redeemScript []byte
+}
+
+func (m *managedP2SHAddress) RedeemScript() (uint16, []byte) {
+	return m.addr.(udb.ManagedScriptAddress).RedeemScript()
 }
 
 // KnownAddress returns the KnownAddress implementation for an address.  The

